src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -263,7 +263,7 @@ internal static async Task SetAppOfflineState(string rootPath, bool offline)
             else if (!offline && offlineFileExists)
             {
                 // delete the app_offline.htm file
-                FileUtility.DeleteFileSafe(path);
+                await Utility.InvokeWithRetriesAsync(() => File.Delete(path), retryIntervalMs: 1000);
```

Comment:
Is it possible that we could have a race here, where we are inside the retry loop but the file has already been deleted (say by another host that received the offline call or by something editing the filesystem directly)? Perhaps it would make more sense to check if the file exists in each retry iteration?

Reply:
Added existence check